ast/Trees.h

```diff
@@ -56,6 +56,23 @@ struct ParsedFile {
     core::FileRef file;
 };
 
+/**
+ * Stores the output of an AST pass. May be empty if pass was canceled or encountered an error.
+ * TODO: Modify to store reason if we ever have multiple reasons for a pass to stop. Currently, it's only empty if the
+ * pass is canceled in LSP mode.
+ */
+class MaybeASTPassResult final {
```

Comment:
Happy to change the name of this.

Reply:
`Maybe` and `Result` sound redundant to me. Maybe pick your favorite.